client/ayon_resolve/plugins/create/create_shot_clip.py

```diff
@@ -240,95 +382,213 @@ def create(self, subset_name, instance_data, pre_create_data):
                 # from `PublishableClip.convert`
                 continue
 
-            track_item.SetClipColor(constants.publish_clip_color)
-
-            instance_data = copy.deepcopy(instance_data)
-            # TODO: here we need to replicate Traypublisher Editorial workflow
-            #  and create shot, plate, review, and audio instances with own
-            #  dedicated plugin
+            self.log.info(
+                "Processing track item data: {} (index: {})".format(
+                    track_item_data, index)
+            )
 
-            # Create the Publisher instance
-            instance = CreatedInstance(
-                product_type=self.product_type,
-                product_name=publish_clip.product_name,
-                data=instance_data,
-                creator=self
+            # Delete any existing instances previously generated for the clip.
+            prev_tag_data = lib.get_timeline_item_ayon_tag(track_item)            
+            if prev_tag_data:
+                for creator_id, inst_data in prev_tag_data[_CONTENT_ID].items():
+                    creator = self.create_context.creators[creator_id]
+                    prev_instances = [
+                        inst for inst_id, inst 
+                        in self.create_context.instances_by_id.items()
+                        if inst_id == inst_data["instance_id"] 
+                    ]
+                    creator.remove_instances(prev_instances)
+
+            # Create new product(s) instances.
+            clip_instances = {}
+            shot_creator_id = "io.ayon.creators.resolve.shot"
+            for creator_id in enabled_creators:
+                sub_instance_data = copy.deepcopy(instance_data)
+                shot_folder_path = sub_instance_data.pop("target_folder_path")
+
+                # Shot creation
+                if creator_id == shot_creator_id:
+                    sub_instance_data.update({
+                        "creator_attributes": {
+                            "folderPath": shot_folder_path,
+                            "workfile_start_frame": \
+                                sub_instance_data["workfileFrameStart"],
+                            "handle_start": sub_instance_data["handleStart"],
+                            "handle_end": sub_instance_data["handleEnd"]
+                        }
+                    })
+
+                # Plate, Audio
+                # insert parent instance data to allow
+                # metadata recollection as publish time.
+                else:
+                    parenting_data = clip_instances[shot_creator_id]
+                    sub_instance_data.update({
+                        "parent_instance_id": parenting_data["instance_id"],
+                        "creator_attributes": {
+                            "folderPath": shot_folder_path,
+                            "parent_instance": parenting_data["label"],
```

Comment:
Lets convert this into `TextDef` with nice label. But let's also make sure this field in non editable by adding input attribute `disabled=True`. For reference look into `ayon-core\client\ayon_core\lib\attribute_definitions.py`

Reply:
Lets actually do the same for the linked child product instances:
![image](https://github.com/user-attachments/assets/fb71671a-cf38-448f-a58a-5f42f13916ef)

I really like how TP is having this done via shared clip base class and the TextDef is added to the class methon `def get_instance_attr_defs`. Here is some reference: [ayon_traypublisher/plugins/create/create_editorial.py#L90-L119](https://vscode.dev/github/ynput/ayon-traypublisher/blob/develop/client/ayon_traypublisher/plugins/create/create_editorial.py#L90-L119)

Reply:
linking here the TP pr for above suggestions https://github.com/ynput/ayon-traypublisher/pull/12

Reply:
Sounds good to me, thanks for the links. Here is what I obtain :

For Plate / Audio:
![image](https://github.com/user-attachments/assets/42d27f9b-356a-496b-a3d4-b99927ad0eb9)
Is the TP "Review" attribute also needed here ? again apologies but Im not entirely sure about this one.

For Shots:
![image](https://github.com/user-attachments/assets/39267df4-ca3e-4732-ae48-c63272aaf9ef)

I've left out the clipIn, clipDuration, clipOut.. read only attributes This way there is no need to query/compute all of this information but maybe that's wrong.. 
My thought was that unlike an EDL import you can visually assess your clips directly from the timeline. 
Is that all right for you ?

Reply:
> I've left out the clipIn, clipDuration, clipOut.. read only attributes This way there is no need to query/compute all of this information but maybe that's wrong..

I believe a user may want to compare and review the data in AYON to ensure its accuracy. This could save us processing time, but it might also increase the time users spend on validation, and so do we during development. I suggest we add clip and frame attributes in a read-only format to streamline this process.

Reply:
> I've left out the clipIn, clipDuration, clipOut.. read only attributes This way there is no need to query/compute all of this information but maybe that's wrong..

If we just had "Instance" **actions** in the Publisher UI we could have a "Select clip" action. 🎉

Reply:
> Is the TP "Review" attribute also needed here ?

Absolutely, that's a valid point. In Resolve or Hiero, we'll have to approach things differently compared to TP. TP has a single track where plates can be flagged for review. For Resolve and Hiero, we should introduce an editable dropdown field on the hero track clips. The dropdown options should come from the names of the timeline tracks. This value is initially set during the creation phase, but users may need to adjust it later if they forget.

Additionally, we'll need to implement a new toggle called "Hero Clip Enabled" (or something along those lines). This feature is crucial for managing vertically aligned clips and is another setting users might need to modify at a later stage.

Reply:
Here is what I got for Shot:
![image](https://github.com/user-attachments/assets/29629309-b9ef-47ff-adc1-2623dc9d859a)

for plate:
![image](https://github.com/user-attachments/assets/0832c793-7e97-4d79-8da5-361d0af2fbcc)

(note that those toggle have currently *no affect* on the plate creation, I think I'd need a meeting or a call with you @jakubjezek001 if more work is expected for this PR about the review product)

and audio product:
![image](https://github.com/user-attachments/assets/f3196fe3-b8f1-44ab-95d2-0b4e56a40bc9)


Does this work for you guys ?